File: backend/app/schemas/blog.py
from pydantic import BaseModel
from datetime import datetime


class BlogBase(BaseModel):
    Title: str
    Slug: str
    Summary: str
    Content: str
    Image: str
    Author: str
    Status: str = "Draft"


class BlogCreate(BlogBase):
    pass


class BlogUpdate(BlogBase):
    pass


class BlogResponse(BlogBase):
    BlogID: int
    CreatedAt: datetime

    class Config:
        from_attributes = True

File: frontend/types/blog.ts
export interface Blog {
    BlogID: number;
    Title: string;
    Slug: string;
    Summary: string;
    Content: string;
    Image: string;
    Author: string;
    Status: string;
    CreatedAt: string;
  }
  
  export interface BlogCreate {
    Title: string;
    Slug: string;
    Summary: string;
    Content: string;
    Image: string;
    Author: string;
    Status: string;
  }
  
  export interface BlogUpdate {
    Title: string;
    Slug: string;
    Summary: string;
    Content: string;
    Image: string;
    Author: string;
    Status: string;
  }